### PR TITLE
add cmd for decode table mvcc data

### PR DESCRIPTION
**Pull request by crazycs520**
you can See doc: https://github.com/pingcap/docs-cn/pull/1197  about how to usage.

```shell
▶ ./tidb-ctl base64decode
Error: Only support 1 or 2 argument!
Usage:
  tidb-ctl base64decode [flags]

Examples:
tidb-ctl base64decode [base64_data]
tidb-ctl base64decode [db_name.table_name] [base64_data]
tidb-ctl base64decode [table_id] [base64_data]

Flags:
  -h, --help   help for base64decode

Global Flags:
  -H, --host ip         TiDB server host (default 127.0.0.1)
  -i, --pdhost ip       PD server host (default 127.0.0.1)
  -p, --pdport uint16   PD server port (default 2379)
  -P, --port uint16     TiDB server port (default 10080)

Only support 1 or 2 argument!

▶ ./tidb-ctl base64decode 58 CAIIAggEAhjlk4jlk4ggaGVsbG8IBgAICAmAgICI0Yyr0Rk=
a:      1
b:      哈哈 hello
c is NULL
d:      2019-03-22 06:20:17
e not found in data

▶ ./tidb-ctl base64decode test.t CAIIAggEAhjlk4jlk4ggaGVsbG8IBgAICAmAgICI0Yyr0Rk=
a:      1
b:      哈哈 hello
c is NULL
d:      2019-03-22 06:20:17
e not found in data

```

### CODE REVIEW

File: cmd/decoderTableMvcc.go [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+e not found in data
+
+As you can see, data of column c is NULL, and data of column e is not found in data, because e is added latter, TiDB currently have not back fill data for added after column.
+
+`
+
+// schemaRootCmd represents the schema command
+var decodeTableCmd = &cobra.Command{
+	Use:     "decodeTable",
+	Short:   "Decode table mvcc data",
```
Comment (jackysp) [inline cmd/decoderTableMvcc.go]: I think it is data. The mvcc data sometimes mean the info of mvcc.

File: cmd/decoderTableMvcc.go [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+		if errClose := resp.Body.Close(); errClose != nil && err == nil {
+			err = errClose
+		}
+	}()
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return
+	}
+	if resp.StatusCode != http.StatusOK {
+		return nil, errors.Errorf("get  table info status code is no ok. body: %s", string(body))
```
Comment (jackysp) [inline cmd/decoderTableMvcc.go]:
```suggestion
		return nil, errors.Errorf("get table info status code is no ok. body: %s", string(body))
```

**Review by jackysp — COMMENTED** [review]
I think we should merge this with `base64decode`. Both of them are decoders for base64 values.

File: doc/tidb-ctl_decodeTable.md [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+### Examples
+
+```
+
+**prepare execute below sql**:
+
+use test;
+create table t (a int, b varchar(20),c datetime default current_timestamp , d timestamp default current_timestamp);
+insert into t (a,b,c) values(1,"陈霜 hello",NULL);
```
Comment (jackysp) [inline doc/tidb-ctl_decodeTable.md]: Maybe you need to `make doc` again. I think the examples are updated.
陈霜 -> 哈哈

File: cmd/decoderTableMvcc.go [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+	Use:     "decodeTable",
+	Short:   "Decode table mvcc data",
+	Long:    "'tidb-ctl decodeTable' decode table mvcc data.",
+	Example: decodeTableExample,
+	RunE:    decodeTableMVCC,
+}
+
+func decodeTableMVCC(_ *cobra.Command, args []string) error {
+	if len(args) != 2 {
+		return errors.Errorf("need 2 param. eg: tidb-ctl decodeTable test.t CAIIAggEAhjpmYjpnJwgaGVsbG8IBgAICAmAgICg9oqr0Rk=")
```
Comment (jackysp) [inline cmd/decoderTableMvcc.go]: If this tool is used by the user, I think it is better not to use `CAIIAggEAhjpmYjpnJwgaGVsbG8IBgAICAmAgICg9oqr0Rk=`.

Comment (crazycs520) [reply cmd/decoderTableMvcc.go]: done.

File: doc/tidb-ctl_decodeTable.md (hunk 3 above)
Comment (crazycs520) [reply doc/tidb-ctl_decodeTable.md]: done.

File: cmd/base64decode.go [hunk 5]
```diff
@@ -51,3 +69,96 @@ func base64decodeCmd(c *cobra.Command, args []string) error {
...
+func getTableInfo(id string) (tblInfo *model.TableInfo, err error) {
+	url := ""
+	if strings.Contains(id, ".") {
+		fields := strings.Split(id, ".")
+		if len(fields) != 2 {
+			return nil, errors.Errorf("wrong table name. need like: test.t1")
+		}
+		url = "/" + fields[0] + "/" + fields[1]
+	} else {
+		// tread as table id.
```
Comment (jackysp) [inline cmd/base64decode.go]:
```suggestion
		// treat as table id.
```

**Review by jackysp — COMMENTED** [review]
LGTM

**Review by lysu — APPROVED** [review]
LGTM

**Pull request merged**